src/main/scala/tech/cryptonomic/conseil/Conseil.scala

```diff
@@ -43,9 +43,15 @@ object Conseil extends App with LazyLogging with EnableCORSDirectives {
           pathPrefix("tezos") {
             Tezos(system.dispatchers.lookup("akka.tezos-dispatcher")).route
           }
-        } ~ logRequest("Service route", Logging.DebugLevel) {
-          pathPrefix("metadata") {
-            PlatformDiscovery(conf)(system.dispatchers.lookup("akka.tezos-dispatcher")).route
+        } ~ pathPrefix("v2") {
+          logRequest("Metadata Route", Logging.DebugLevel) {
+            pathPrefix("metadata") {
+              Metadata(conf)(system.dispatchers.lookup("akka.tezos-dispatcher")).route
```

Comment:
I'm just realizing that this routes may not be memoized between calls... or are them?
Not sure about akka-http routing semantic... here

If that's the case, we better pre-instantiate all routes that don't depend on query-specific params to external lazy vals, and then refer to the vals within the route dsl builder

Reply:
I think they are generated at the beginning, but I cannot find confirmation. It would be silly if the routes could change during runtime.

Reply:
The route tree definition may not change, but the underlying logic might well do:

Look [here to be convinced](https://doc.akka.io/docs/akka-http/current/routing-dsl/index.html#dynamic-routing-example)

>>>As the routes are evaluated for each request, it is possible to make changes at runtime. Please note that every access may happen on a separated thread, so any shared mutable state must be thread safe.

I'm worried by the fact that Directives takes a thunk to evaluate a response, so each time the code matches for, e.g.
```scala
pathPrefix("metadata") {
  code...
}
```
the `code` will be executed each time, rebuilding the sub-routes each time, possibly creating complex and heavy-weight objects...

Reply:
it would if I used it this way, but I'm not - each call is either to the object or to the injected class so I don't think it is a problem

Reply:
by injected class I mean the object created out of class and injected to the route